test/features/environment.py

```diff
@@ -139,6 +142,48 @@ def before_scenario(context, scenario):
     # Set personas.
     context.personas = PERSONAS
 
+    for tag in scenario.tags:
+        FIXTURE_NAME = 0
+        PARAMS = slice(1, None)
+
+        parts = tag.split("::")
+
+        if parts[FIXTURE_NAME].startswith("fixture.dataset_with_schema"):
+            use_fixture(dataset_with_schema, context, parts[PARAMS])
+        elif parts[FIXTURE_NAME].startswith("fixture.create_resource_for_dataset_with_params"):
+            use_fixture(create_resource_for_dataset_with_params, context, parts[PARAMS])
+
 
 def after_scenario(context, scenario):
     benv.after_scenario(context, scenario)
+
+
+@fixture
+def dataset_with_schema(context, path="", **kwargs):
+    params = "&".join(path)
+    context.execute_steps(u"""
+        Given browser "remote"
+        Then I visit "api/action/qld_test_create_dataset?{}"
+    """.format(params))
+
+    json_content = context.browser.find_by_tag("pre")[0].text
+    pkg_data = json.loads(json_content)['result']
+    pkg_id = pkg_data['id']
+    context.dataset = pkg_data
+
+    yield
+
+    context.execute_steps(u"""
+        Given browser "remote"
+        Then I visit "api/action/qld_test_purge_dataset?id={}"
```

Comment:
I don't agree to trying to load/purge data this way, 'create-test-data.sh' is there for a reason for static test data. 

You can have tests generate a uuid string name that is used further in said test. You should not be using the same 'dataset' on multiple tests if not created prior to testing due to out of order or multi container forked test runs in parallel that could occur in the future.